### PR TITLE
Replace Gulp with Webpack

### PR DESCRIPTION
**Pull request by synbot**
### ACs
1. Gulp is gone
1. Webpack builds app.js file
1. Webpack builds app.css file
1. Webpack rebuilds js and css without reloading the page
1. Webpack builds media files
1. Webpack shows js and jsx warnings
1. Tests build using webpack
1. Tests use the hot loader
1. Environment variable replacing works
1. Sourcemaps work
1. JShint and JSXhint run on appropriate files
1. Webpack and Hot Loader start with `npm start`
1. Browser tests can be run with `npm test`
1. Karma test (command line executed in Chrome and Firefox) can be run with `npm run karma`
1. Production builds can be created with `npm run dist`

### Notes
- Biggest difference from the gulp build is that media files will be included with their full path

### CODE REVIEW

**Comment by areida** [comment]
I think this is pretty much good to go. One issue is that the webpack-dev-server doesn't serve any path except `/`. I think this can be fixed using webpack-dev-middleware and a node server but that change is pore appropriate for #70.

**Comment by paulstatezny** [comment]
:three: minimum, not 0.

**Comment by paulstatezny** [comment]
:o:

**Comment by areida** [comment]
:star:

File: __tests__/index.js [hunk 1]
```diff
@@ -1,7 +1,13 @@
+/* jshint expr: true, globalstrict: true */
 'use strict';
+
 // Ensure that chai accepts should-style assertions and is integrated with sinon.
 var chai      = require('chai');
 var sinonChai = require('sinon-chai');
 
-chai.should();
```
Comment (paulstatezny) [inline __tests__/index.js]: @zpchavez This was causing infinite loops ("max stack call size exceeded"). Any idea?

Comment (zpchavez) [inline __tests__/index.js]: No. I imagine we've already tried updating chai and/or sinon-chai?

**Comment by baohx2000** [comment]
:pray:

**Comment by paulstatezny** [comment]
And a resounding :+1:

**Pull request merged**